FWCore/Framework/test/stubs/TestStreamAnalyzers.cc

```diff
@@ -220,23 +207,20 @@ namespace edmtest {
                                            LuminosityBlockContext const* iLBContext) {
         ++m_count;
         auto pCache = iLBContext->luminosityBlock();
-        if (pCache->lumi != 1) {
+        if (pCache->run != iLB.luminosityBlockAuxiliary().run() ||
+            pCache->lumi != iLB.luminosityBlockAuxiliary().luminosityBlock()) {
           throw cms::Exception("end out of sequence")
               << "globalEndLuminosityBlock seen before globalBeginLuminosityBlock in LuminosityBlock"
               << iLB.luminosityBlock();
         }
-        gel = true;
-        gbl = false;
```

Comment:
Setting it to 0 at globalEndRun is harmless, but serves no purpose. A new Cache is constructed for every run and the old ones are deleted. I removed this line from RunIntAnalyzer.

Reply:
Setting it zero used to serve a purpose because it was only incrementing Cache::run to 1 for all runs. So if there was some problem with globalBeginRun and a new Cache was not created the zero would be noticed, but now I'm actually putting the run number in the cache and run number changes so zeroing it is not useful.

Reply:
It seemed to me the zeroing in `globalEndRun()` would serve a purpose of catching a case where `globalEndRun()` would be run before (stream)  `endRun()`, because the `pCache->run` would no longer match the run number in that case.

Reply:
You convinced me. Adding the ones you suggested now and putting the one I removed back.